File: python/src/two_sum.py
# encoding: utf-8

"""
no.: 1
url: https://leetcode.cn/problems/two-sum/
"""

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable = dict()
        for i, num in enumerate(nums):
            if target - num in hashtable:
                return [hashtable[target - num], i]
            hashtable[nums[i]] = i

        return []

File: rust/src/two_sum.rs
fn main() {
    let nums = [2, 7, 11, 15, 23];
    let target = 9;
    let mut hashtable = std::collections::HashMap::<i32, i32>::new();
    for i in 0..nums.len() {
        if hashtable.contains_key(&(target - nums[i])) {
            println!("{}, {}", nums[i], i);
            return [nums[i], i];
        }
        hashtable.insert(nums[i], i);
    }
    return [];
}
